sktime/forecasting/base/tests/test_fh.py

```diff
@@ -49,7 +53,10 @@ def test_fh(index_type, fh_type, is_relative, steps):
     """Testing ForecastingHorizon conversions."""
     # generate data
     y = make_forecasting_problem(index_type=index_type)
-    assert isinstance(y.index, INDEX_TYPE_LOOKUP.get(index_type))
+    if index_type == "int":
```

Comment:
confused. Instead of this if/else construction, why do you not just write 
`assert is_in_valid_index_types(y.index)`?

Reply:
```
assert isinstance(y.index, INDEX_TYPE_LOOKUP.get(index_type))
```
This line checks for a very specific index type, not just the set of valid indices. Since the index that corresponds to `int` key in `INDEX_TYPE_LOOKUP` is very generic `pd.Index` (instead of `pd.Int64Index`) I have to check that it is indeed integer index. Hence, this ugly `if...else`.

Reply:
oh, I seel. How about giving `is_in_valid_index_types` an optional second argument, from the lookup dictionary then? Then it checks only for that string.

Reply:
Sorry, but at this point I will start to resist. This proposal moves complexity from the test to the source code of the library. We already have a few specific and super simple functions to check for specific index types: `is_integer_index`, `is_in_valid_relative_index_types`, and `is_in_valid_absolute_index_types`. Refactoring those once more to something with optional arguments just to accommodate one single test does not sound very attractive to me.

Reply:
> from the test to the source code

Ah, that´s a valid point!
It´s a *test*, not the library code!
Ok, I´m fine with the status quo then.